process/sync/baseSync.go

```diff
@@ -118,33 +135,35 @@ func (boot *baseBootstrap) loadBlocks(
 		}
 	}
 
-	if currentNonce <= blockFinality {
-		err = process.ErrNotEnoughValidBlocksInStorage
-		lastNotarized = make(map[uint32]uint64, 0)
-		finalNotarized = make(map[uint32]uint64, 0)
-		currentNonce = 0
-	}
-
-	if err == nil {
-		err = applyNotarizedBlocks(finalNotarized, lastNotarized)
+	defer func() {
 		if err != nil {
 			lastNotarized = make(map[uint32]uint64, 0)
 			finalNotarized = make(map[uint32]uint64, 0)
-			currentNonce = 0
+			validNonce = 0
 		}
-	}
 
-	if err == nil {
-		boot.addHeadersToForkDetector(shardId, currentNonce, getHeader)
+		for i := validNonce + 1; i <= highestNonceInStorer; i++ {
```

Comment:
This does redundant job. couldn't have been from validNonce+1 to current nonce (before current nonce was equaled to valid nonce)?

Reply:
We should remove all the blocks above the validNonce at the end